### DISCUSSION TITLE
2fa changed, but I didn't change it...??

### DISCUSSION THREAD

**Discussion by jerryasher** [community/community#55790] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I tried logging in today on my phone and it failed when I was told my 2fa code was wrong. I tried to enter the code several times waiting for it to cycle to new codes -- they all failed.

That's intriguing!

I eventually got back in using one of the backup/recovery mechanisms, but examining the security log, I see no reason why the 2fa code was rejected. Last month, 30 days ago, I used the same 2fa key to log in successfully to connect an app to my account, and there are no intervening 2fa changes.  Hence, I recognize all of the security log entries between now and the last time I used the authenticator app successfully.

Anyone have any clue as to what might have happened?

This is the complete security log between my unsuccessfully login attempt an hour ago with a failed 2fa code and 30 days ago when I used 2fa successfully to login

```
jerryasher – user.two_factor_challenge_failure
Two-factor challenge failed with authenticator app
10-.---.---.--- San Francisco, California, United States 1 hour ago

jerryasher – user.two_factor_requested
Two-factor auth requested
10-.---.---.--- San Francisco, California, United States 1 hour ago

jerryasher – user.new_device_used
New Device Used
10-.---.---.--- San Francisco, California, United States 1 hour ago

jerryasher – oauth_authorization.create
Created authorization for OAuth application (BIGNAMEAPPCO) with scope(s)
10-.---.---.--- San Francisco, California, United States last month

jerryasher – user.login
Logged in
10-.---.---.--- San Francisco, California, United States last month

jerryasher – user.sign_in_from_unrecognized_device
User signed in from an unrecognized device.
10-.---.---.--- San Francisco, California, United States last month

jerryasher – user.two_factor_challenge_success
Two-factor auth challenge solved with authenticator app
10-.---.---.--- San Francisco, California, United States last month

jerryasher – user.two_factor_requested
Two-factor auth requested
10-.---.---.--- San Francisco, California, United States last month

jerryasher – user.new_device_used
New Device Used
10-.---.---.--- San Francisco, California, United States last month
```

**Comment by airtower-luna** [comment]
The first thing to check is time. TOTP is time-based (the first T), so if your phone clock is out of sync the codes won't match.

**Reply by jerryasher** [reply]
Thanks, but it's doubtful it's a time issue. My TOTP auth generator is on my internet connected android phone, and checking it's internal clock right now against https://greenwichmeantime.com/ the two clocks seem almost identically synced. And the very small, less than a second, visual lag there is on the phone would not explain why multiple attempts at generating a code, failing, then waiting for a new cycle and trying again all failed.

**Comment by seantleonard** [comment]
I just had the same issue where my once working TOTP code from Microsoft Authenticator was working, and without any changes, all my codes where failing. I just had to update my TOTP account with a new QR code to get things working again. Frustrating.

**Comment by irank786** [comment]
Whether you need a 24/7 emergency[ locksmith in Fremont,](https://access24locksmith.com) lock maintenance for your properties, or want to upgrade to smart and high security lock, our mobile locksmiths are ready to help, same day.
